crates/pico/tests/try_ok_split.rs

```diff
@@ -0,0 +1,34 @@
+use pico::Storage;
+use pico_macros::{Db, memo};
+
+#[derive(Db, Default)]
+struct TestDatabase {
+    storage: Storage<Self>,
+}
+
+#[test]
+fn try_ok_then_split_projects_each_tuple_value() {
+    let db = TestDatabase::default();
+
+    let (left, right) = ok_pair_value(&db)
+        .try_ok()
+        .expect("expected Ok result before split")
+        .split();
+
+    let _ = &*left;
+    let _ = &*right;
+}
+
+#[memo]
+fn ok_pair_value(_db: &TestDatabase) -> Result<(PanicOnClone, PanicOnClone), ()> {
```

Comment:
QQ: what need do we have for tuples? Is that a specific, concrete thing we need?